api/src/helpers/origin.ts

```diff
@@ -0,0 +1,36 @@
+import { GraphQLError } from 'graphql';
+import wildcard from 'wildcard-match';
+
+export function isOriginValid(originParam: string): boolean {
+  if (originParam === '*') {
+    return true;
+  }
+  if (originParam.includes(' ')) {
+    return false;
+  }
+  const origins = originParam.split(',');
+  return origins.some((origin) => {
+    if (!origin.startsWith('http://') && !origin.startsWith('https://')) {
+      return false;
```

Comment:
Instead of returning true or false, validators should normally return error message that let the user understand what's wrong (with enough information to fix it)